lib/services/plugins-service.ts

```diff
@@ -43,14 +50,62 @@ export class PluginsService implements IPluginsService {
 
 	public remove(pluginName: string): IFuture<void> {
 		return (() => {
+			let isUninstallCommandExecuted = false;
+
 			let removePluginNativeCodeAction = (modulesDestinationPath: string, platform: string, platformData: IPlatformData) => {
-				let pluginData = this.convertToPluginData(this.getNodeModuleData(pluginName).wait());
-				pluginData.isPlugin = true;
-				return platformData.platformProjectService.removePluginNativeCode(pluginData);
+				return (() => {
+					let pluginData = this.convertToPluginData(this.getNodeModuleData(pluginName).wait());
+
+					platformData.platformProjectService.removePluginNativeCode(pluginData).wait();
+
+					// Remove the plugin and call merge for another plugins that have configuration file
+					let pluginConfigurationFilePath = this.getPluginConfigurationFilePath(pluginData, platformData);
+					if(this.$fs.exists(pluginConfigurationFilePath).wait()) {
+						let tnsModulesDestinationPath = path.join(platformData.appDestinationDirectoryPath, constants.APP_FOLDER_NAME, constants.TNS_MODULES_FOLDER_NAME);
+						let nodeModules = this.$broccoliBuilder.getChangedNodeModules(tnsModulesDestinationPath, platform).wait();
+						let plugins = _.map(_.keys(nodeModules), nodeModule => this.convertToPluginData(this.getNodeModuleData(pluginName).wait()));
+
+						_.forEach(plugins, (data, index) => {
+							if(data.isPlugin) {
+								if(this.$fs.exists(this.getPluginConfigurationFilePath(data, platformData)).wait()) {
+									if(!isUninstallCommandExecuted) {
+										this.executeNpmCommand(PluginsService.UNINSTALL_COMMAND_NAME, pluginName).wait();
+										isUninstallCommandExecuted = true;
+									}
+
+									if(index === 0) {
+										this.$projectDataService.initialize(this.$projectData.projectDir);
+										let frameworkVersion = this.$projectDataService.getValue(platformData.frameworkPackageName).wait().version;
+										this.$npm.cache(platformData.frameworkPackageName, frameworkVersion).wait();
+
+										let relativeConfigurationFilePath = path.relative(platformData.projectRoot, platformData.configurationFilePath);
+										let cachedPackagePath = this.$injector.resolve("npmInstallationManager").getCachedPackagePath(platformData.frameworkPackageName, frameworkVersion);
+										let cachedConfigurationFilePath = path.join(cachedPackagePath, constants.PROJECT_FRAMEWORK_FOLDER_NAME, relativeConfigurationFilePath);
+
+										shelljs.rm("-rf", platformData.configurationFilePath);
```

Comment:
Removing the old configuration file seems redundant. If it is not removed it'll be replaced by the force `shell.cp` below